### PR TITLE
Consistify zlib documentation

### PR DESCRIPTION
**Pull request by jeffpc**
This updates the docs to be consistent with each other and reality.

FWIW, I looked the code in core and there are some interesting assumptions about what values are valid for levels.  I have some WIP code to make that stuff sane, but that'll have to wait for next week at the very least.  So, this PR should be considered stand-alone.

### CODE REVIEW

File: source/configuration_manual/mail_location/obox/compression.rst [hunk 1]
```diff
@@ -41,10 +41,10 @@ compressed to fscache. On one hand this increases the fscache's file, but on
 the other hand it requres Dovecot to always uncompress the files before
 accessing them.
 
-This uncompression means that the file is temporarily written to
-``mail_temp_dir``. By using the ``compress`` fs wrapper in ``obox_fs`` line the
-mails are stored uncompressed in fscache, and reading the mails from there
-doesn't require writing to ``mail_temp_dir``.
+This decompression uses a temporary that is written to ``mail_temp_dir``. By
```
Comment (sirainen) [inline source/configuration_manual/mail_location/obox/compression.rst:44]: "temporary file"

File: source/configuration_manual/mail_location/obox/compression.rst [hunk 2]
```diff
@@ -41,10 +41,10 @@ compressed to fscache. On one hand this increases the fscache's file, but on
...
+using the ``compress`` fs wrapper in ``obox_fs`` line the mails are stored
+uncompressed in fscache, and reading the mails from there doesn't require
```
Comment (sirainen) [inline source/configuration_manual/mail_location/obox/compression.rst:46]: This is only true if compress is specified after fscache, not true if before. Should also clarify that here now that it's being changed anyway..

File: source/configuration_manual/zlib_plugin.rst [hunk 3]
```diff
@@ -25,10 +25,19 @@ Configuration:
...
-    zlib_save = gz # or bz2, lz4 or zstd
+    zlib_save = gz
+    zlib_save_level = 6
   }
 
+The ``zlib_save`` setting selects the compression algorithm (currently
+supported values are: gz, bz2, lz4, zstd) to use when saving a new mail.
+The ``zlib_save_level`` setting sets the compression level used.
+
+Note: Currently, the compression level must be an integer in the range 1 to
```
Comment (sirainen) [inline source/configuration_manual/zlib_plugin.rst:36]: Maybe it should be ".. note::"?

**Review by sirainen — CHANGES_REQUESTED** [review]

File: source/configuration_manual/mail_location/obox/compression.rst (hunk 2 above)
Comment (jeffpc) [reply source/configuration_manual/mail_location/obox/compression.rst:46]: Right.  s/wrapper/wrapper after ``fscache``/

**Review by cmouse — APPROVED** [review]

**Review by sirainen — APPROVED** [review]

**Pull request merged**